src/main/java/views/add_edit_category_views/CategoryP.java

```diff
@@ -11,31 +11,30 @@
 public class CategoryP implements CategoryOB {
     /**
      * Overrides success_add from Category_OB.
-     * @param categoryOD_add CategoryOD Object produced from adding the category
      * @return CategoryOD from success add attempt
      */
     @Override
-    public CategoryOD success_add(CategoryOD categoryOD_add) {
-        // category(OD) update in month
-        return categoryOD_add;
+    public CategoryOD success_add() {
+        String success_add = "You have added a new category!";
+        return new CategoryOD(success_add);
     }
     /**
      * Overrides fail() from Category_OB.
      * @param error Detailed error message for user.
      * @return String description of error from user.
      */
     @Override
-    public String fail(String error){
-        return error;
+    public CategoryOD fail(String error){
+        return new CategoryOD(error);
     }
     /**
      * Overrides success_edit from Category_OB.
```

Comment:
I think a better description for these overridden methods are needed. The fact that its overridden doesn't matter to the user, they just want to know what it does. In mine i just copied the same descriptions i had in my output boundary.